app/controllers/product_categories_controller.rb

```diff
@@ -0,0 +1,103 @@
+class ProductCategoriesController < ApplicationController
+  helper_method :get_cart_order_products
+
+  def index
+    @product_category = ProductCategory.new
+    @product = Product.new
+    @product_categories
+
+    if Rails.cache.fetch(:product_categories)
+      @product_categories = Rails.cache.fetch(:product_categories)
```

Comment:
can we have a separate util for redis operations and handle these operations there ?